src/device/device.go

```diff
@@ -17,8 +23,49 @@ func (device *Device) String() string {
 	return "Device:{ deviceId:" + strconv.Itoa(device.ID) + ", Host:" + device.Host + ", port:" + device.Port + " }"
 }
 
+//Init - Initialize all the components on this device
+func (device *Device) Init() error {
```

Comment:
You suggest you are going to return an error here but only ever return nil

Reply:
the err return was missing, added